### ISSUE TITLE
Bug: Cast string to float is not supported ImageClassifier

### ISSUE THREAD

**Issue by JuliaWasala** [keras-team/autokeras#1818]
### Bug Reproduction
Code for reproducing the bug:

```
clf =ak.ImageClassifier(overwrite=True,project_name=name,directory="/data1/wasalaj/methane_plumes/models/")
clf.fit(X_train,y_train,validation_data=(X_val,y_val))

print("DONE TRAINING - PREDICT")
predicted_y = clf.predict(X_test)

print("GET SCORES")
# get acc, precision, recall, f1
acc = accuracy_score(y_test, predicted_y)
precision = precision_score(y_test, predicted_y,average='macro')
recall = recall_score(y_test, predicted_y,average='macro')
f1 = f1_score(y_test, predicted_y,average='macro')
print(acc,precision,recall,f1)
```

Data used by the code:
private data, (1,32,32) images. 

### Setup Details
Include the details about the versions of:
 - OS type and version: CentOS 7.9.2009
 - Python: 3.9.13
 - autokeras: 1.0.20
 - keras-tuner: 1.1.3
 - scikit-learn: 1.2.0
 - numpy: 1.23.5
 - pandas: 1.5.2
 - tensorflow: 2.11

**Comment by JuliaWasala** [comment]
update: weight decay is the culprit. Need a fix for this.

**Comment by JuliaWasala** [comment]
#1802 addresses this.
